misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -15,25 +15,129 @@
  */
 package org.ballerinalang.formatter.core;
 
+import io.ballerina.compiler.syntax.tree.ConstantDeclarationNode;
 import io.ballerina.compiler.syntax.tree.ImportDeclarationNode;
 import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.projects.TomlDocument;
+import io.ballerina.toml.semantic.ast.TomlTableNode;
 import io.ballerina.tools.text.LineRange;
 import org.apache.commons.lang3.builder.CompareToBuilder;
+import org.ballerinalang.formatter.core.options.WrappingFormattingOptions;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 /**
  * Class that contains the util functions used by the formatting tree modifier.
  */
-class FormatterUtils {
+public class FormatterUtils {
 
     private FormatterUtils() {
 
     }
 
     static final String NEWLINE_SYMBOL = System.getProperty("line.separator");
 
+    public static Map<String, Object> getFormattingConfigurations(Path root, String path) throws FormatterException {
+        String content;
+        if (isLocalFile(path)) {
+            try {
+                content = Files.readString(Path.of(path), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to retrieve local formatting configuration file");
+            }
+        } else {
+            content = readRemoteFormatFile(root, path);
```

Comment:
The content can be null if this fails to read, right? Can't we produce the error message when this fails similar to how you are doing with local files?

Reply:
These cases are handled in the [readRemoteFormatFile](https://github.com/poorna2152/ballerina-lang/blob/formatter/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java#L84-L121) method.